FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperationForgotPassword.m

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) 2017 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "FUIAccountSettingsOperationForgotPassword.h"
+
+#import "FUIAccountSettingsOperation_Internal.h"
+#import "FUIAuthBaseViewController.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation FUIAccountSettingsOperationForgotPassword
+
+-(void)execute:(BOOL)showDialog {
+  [self onForgotPassword];
+}
+
+- (void)onForgotPassword {
+  __block FUIStaticContentTableViewCell *inputCell =
+  [FUIStaticContentTableViewCell cellWithTitle:FUILocalizedString(kStr_Email)
+                                        value:_delegate.auth.currentUser.email
+                                        action:nil
+                                          type:FUIStaticContentTableViewCellTypeInput];
+  FUIStaticContentTableViewContent *contents =
+      [FUIStaticContentTableViewContent
+           contentWithSections:@[
+                                 [FUIStaticContentTableViewSection sectionWithTitle:nil
+                                                                              cells:@[inputCell]],
+                                ]];
+
+  UIViewController *controller =
+      [[FUIStaticContentTableViewController alloc]
+           initWithContents:contents
+                  nextTitle:FUILocalizedString(kStr_Send)
+                 nextAction:^{ [self onPasswordRecovery:inputCell.value]; }
+                 headerText:FUILocalizedString(kStr_PasswordRecoveryMessage)];
+  controller.title = FUILocalizedString(kStr_PasswordRecoveryTitle);
+  [_delegate pushViewController:controller];
+}
+
+- (void)onPasswordRecovery:(NSString *)email {
+  if (![[FUIAuthBaseViewController class] isValidEmail:email]) {
+    [self showAlertWithMessage:FUILocalizedString(kStr_InvalidEmailError)];
+    return;
+  }
+
+  [_delegate incrementActivity];
+
+  [_delegate.auth sendPasswordResetWithEmail:email
+                             completion:^(NSError *_Nullable error) {
+     // The dispatch is a workaround for a bug in FirebaseAuth 3.0.2, which doesn't call the
```

Comment:
This should be fixed by now. Is there any chance we can make FirebaseUI depend on FirebaseAuth ~> 3.1.0 ?